### PR TITLE
Various performance improvements

### PR DESCRIPTION
**Pull request by moffatman**
Should be twice as fast according to my use cases. Happy to discuss what you need for testing, (micro)benchmarking, preserving public API for old slow functions, or whatever else to merge it. I know I still need to do stuff like bump version, changelog, etc.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

### CODE REVIEW

**Comment by kevmoo** [comment]
please run the formatter!

**Comment by moffatman** [comment]
Idk whether the new `trie.dart` is the right way... maybe something to wait for dart macros

**Review by kevmoo — CHANGES_REQUESTED** [review]
need to run the formatter, it seems

File: lib/src/constants.dart [hunk 1]
```diff
@@ -393,10 +393,54 @@ const mathmlTextIntegrationPointElements = [
   Pair(Namespaces.mathml, 'mtext')
 ];
 
-const spaceCharacters = ' \n\r\t\u000C';
+class Charcode {
```
[the remark this answers is not in the document]
Comment (moffatman) [reply lib/src/constants.dart]: Thx done

**Comment by mosuem** [comment]
Closing as the [dart-lang/html](https://github.com/dart-lang/html) repository is merged into the [dart-lang/tools](https://github.com/dart-lang/tools) monorepo. Please re-open this PR there!

**Pull request closed without merging**